relay-server/src/actors/store.rs

```diff
@@ -846,6 +895,17 @@ struct ReplayRecordingKafkaMessage {
     retention_days: u16,
 }
 
+#[derive(Debug, Serialize)]
+struct ReplayRecordingKafkaMessage {
+    replay_id: EventId,
+    key_id: Option<u64>,
+    org_id: u64,
+    project_id: ProjectId,
+    received: u64,
+    retention_days: u16,
+    payload: Bytes,
```

Comment:
It might simplify the code if we only defined a single type `ReplayRecordingKafkaMessage`, with

```rust
#[serde(skip_serializing_if = "Option::is_none")]
payload: Option<Bytes>,

#[serde(skip_serializing_if = "Option::is_none")]
replay_recording: Option<ChunkedReplayRecording>,
```

I realize this would in theory allow the creation of a message with both fields set, but it would allow you to simplify `produce_replay_recording` to something like

```rust
let mut message = ...;
if item.payload().len() < max_payload_size {
    message.payload = ...;
} else {
    message.replay_recording = ...;
}
metric!(...);
self.produce(...);
```

Reply:
I prefer to keep impossible states impossible.